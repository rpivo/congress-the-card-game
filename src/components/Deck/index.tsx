import React, { memo, useContext, useState } from 'react';
import TakeIcon from './TakeIcon';
import { AppContext } from '@components/App';
import Style from '@components/Card/style';

const Deck = (): JSX.Element => {
  const { setShouldDisplayHand } = useContext(AppContext);
  const [canDrawCard, setCanDrawCard] = useState(true);

  const handleClick = () => {
    setCanDrawCard(false);
    setShouldDisplayHand(true);
  };

  return (
    <Style className='deck'>
      <Style
        className={`stackedCard${canDrawCard ? ' canDrawCard' : ''}`}
        onClick={() => canDrawCard && handleClick()}>
        {canDrawCard && <TakeIcon />}
      </Style>
    </Style>
  );
};

export default memo(Deck);
