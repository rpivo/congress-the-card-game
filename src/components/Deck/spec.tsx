import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Deck from '@components/Deck';

describe('Deck', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Deck />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
