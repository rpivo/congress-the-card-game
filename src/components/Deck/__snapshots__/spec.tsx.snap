// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck should render correctly 1`] = `
.c1 {
  cursor: pointer;
}

.c1.endTurnIcon {
  display: inline-block;
  position: fixed;
  right: 10px;
  text-align: center;
  top: 10px;
}

.c1.endTurnIcon p {
  margin-top: -10px;
}

.c1.endTurnIcon.endable line,
.c1.endTurnIcon.endable path {
  -webkit-animation: pulse 3s infinite;
  animation: pulse 3s infinite;
}

.c0 {
  background: #EAEAEA;
  border-radius: 5px;
  cursor: pointer;
  display: inline-block;
  height: 210px;
  padding: 8px;
  place-self: center;
  position: relative;
  -webkit-transition: box-shadow 0.2s ease-in-out;
  transition: box-shadow 0.2s ease-in-out;
  width: 150px;
}

.c0.active {
  background: #9FFFB0;
}

.c0.card {
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
}

.c0.card:active {
  background: #9FFFB0;
}

.c0.card:hover {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25),0 10px 10px rgba(0,0,0,0.22);
}

.c0.deck {
  background: #C6C6C6;
  grid-row: 3;
  grid-column: 1;
  box-shadow: 0 1px 3px rgba(0,0,0,0.52),0 1px 2px rgba(0,0,0,0.64);
}

.c0.stackedCard {
  margin-left: -11px;
  margin-top: -11px;
}

.c0.stackedCard.canDrawCard:hover {
  background: #9FFFB0;
}

.c0.stackedCard.canDrawCard:hover .drawCardIcon circle {
  fill: #353;
}

.c0.stackedCard.canDrawCard:hover .drawCardIcon p {
  margin-top: 7px;
}

.c0.stackedCard.canDrawCard:hover .drawCardIcon path,
.c0.stackedCard.canDrawCard:hover .drawCardIcon line {
  stroke: #9FFFB0;
}

.c0.stackedCard.canDrawCard:hover .drawCardIcon svg {
  width: 56;
  height: 56;
  margin-top: -3px;
}

.c0.stackedCard .drawCardIcon {
  margin-top: 60px;
  text-align: center;
}

.c0.stackedCard .drawCardIcon p {
  margin-top: 10px;
}

.c0 p {
  margin-bottom: 0;
  margin-top: 0;
}

.c0 p.subtext {
  font-size: 11px;
  margin-top: 3px;
}

.c0 p.ability1Title {
  margin-top: 15px;
}

.c0 .xIcon {
  -webkit-align-itmes: center;
  -webkit-box-align-itmes: center;
  -ms-flex-align-itmes: center;
  align-itmes: center;
  background: #FF0000;
  border-radius: 100%;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 18px;
  height: 20px;
  width: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 18px;
  padding: 2px;
  position: absolute;
  text-align: center;
  top: -6px;
  right: -6px;
}

@media screen and (max-width:1750px) {
  .c1.endTurnIcon {
    right: 22px;
    top: 22px;
  }
}

<div
  className="c0 deck"
>
  <div
    className="c0 stackedCard canDrawCard"
    onClick={[Function]}
  >
    <div
      className="c1 drawCardIcon"
      onClick={[Function]}
    >
      <svg
        height="50"
        viewBox="0 0 703 703"
        width="50"
      >
        <g
          transform="translate(-398 -203)"
        >
          <circle
            cx="351.5"
            cy="351.5"
            fill="#707070"
            r="351.5"
            transform="translate(398 203)"
          />
          <path
            d="M136,136,0,272ZM0,0,136,136Z"
            fill="none"
            stroke="#EAEAEA"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="100"
            transform="translate(805.5 418.5)"
          />
          <line
            fill="none"
            stroke="#EAEAEA"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="100"
            transform="translate(556.5 554.5)"
            x2="385"
          />
        </g>
      </svg>
      <p>
        Take a Card
      </p>
    </div>
  </div>
</div>
`;
